backend/routes/graph.py

```diff
@@ -300,33 +302,48 @@ def dfs(node, depth):
     return tree
 
 
-# TODO: Make test case to reflect diagram - https://app.diagrams.net/#G1mIthDUn4T1y1G3BdupdYKPkZVyQZ5XYR
-#  - for parents at top not in graph:, label like such. if one in the graph is '1', parents out of graph are 1p1, 1p2.
-#  if want further ancestry, 1p1p1, etc.
-# test code for the above:
-# G = nx.DiGraph([('a','b'), ('a','c'), ('b','d'), ('b','e'), ('c','f'), ('2', 'c'), ('1', '2'), ('1', 'a')])
-# target_nodes = ['d', 'e', 'f']
-# assert connect_roots(G, target_nodes).edges == nx.DiGraph([('a','b'), ('a','c'), ('b','d'), ('b','e'), ('c','f')]).edges
-# TODO: make sure it found missing in between nodes 5, 7, and 17 and that this is the only diff
-# print(list(connect_roots(G, target_nodes).edges))
-
-# test tree paths:
-# G = nx.DiGraph([('a','b'), ('a','c'), ('b','d'), ('b','e'), ('c','f'), ('2', 'c'), ('1', '2'), ('1', 'a')])
-# sg = G.subgraph(['a', 'b', 'c', 'd', 'e', 'f', '1', '2'])
-# assert set(sg.edges) == set([('a','b'), ('a','c'), ('b','d'), ('b','e'), ('c','f'), ('2', 'c'), ('1', '2'), ('1', 'a')])
-# assert tree_paths = get_indented_tree_nodes(sg) == [ (0, '1'), (1, '2'), (2, 'c'), (3, 'f'), (1, 'a'), (2, 'b'), (3, 'd'), (3, 'e'), (2, 'c'), (3, 'f'), (3, 'c') ]
-# assert print_tree_paths(tree_paths) == """
-# 1
-#     2
-#         c
-#             f
-#     a
-#         b
-#             d
-#             e
-#         c
-#             f
-# """
+nodes_from_edges = lambda edges : set(x for edge in edges for x in edge)
```

Comment:
## `lambda` `nodes_from_edges`
I know lambdas are often discouraged, but w/e. 

It works correctly, but don't you find that the English interpretation of this just doesn't seem to make sense?: ` set(x for edge in edges for x in edge)`. Unusual for Python.